ci/docker/linux-apt-r.dockerfile

```diff
@@ -103,13 +103,18 @@ ENV \
     ARROW_BUILD_STATIC=OFF \
     ARROW_BUILD_TESTS=OFF \
     ARROW_BUILD_UTILITIES=OFF \
+    ARROW_COMPUTE=ON \
+    ARROW_CSV=ON \
+    ARROW_DATASET=ON \
+    ARROW_FILESYSTEM=ON \
     ARROW_FLIGHT=OFF \
     ARROW_GANDIVA=OFF \
+    ARROW_HDFS=ON \
+    ARROW_JSON=ON \
```

Comment:
Is all this required for R? AFAIR, Python is only used to test PyArrow-R interoperability here.

Reply:
@nealrichardson Do you know what components are needed here?

Reply:
It seems that we can disable `ARROW_HDFS`. Other components seems necessary. I'll try it.